templates/release.yml

```diff
@@ -0,0 +1,102 @@
+parameters:
+  ServiceConnection:
+
+steps:
+    - task: AzurePowerShell@5
+      displayName: 'Show Variables'
+      inputs:
+        azureSubscription: ${{ parameters.ServiceConnection }}
+        ScriptType: InlineScript
+        Inline: |
+          # Setting Variables
+          Write-Output $(KeyVaultId)
+          Write-Output $(dockerRegistry)
+          Write-Output $(dockerImageName)
+          Write-Output $(Build.BuildNumber)
+          write-output ${{ parameters.ServiceConnection }}
+        preferredAzurePowerShellVersion: 3.1.0
+
+    - task: charleszipp.azure-pipelines-tasks-terraform.azure-pipelines-tasks-terraform-cli.TerraformCLI@0
+      displayName: 'Terraform Init'
+      inputs:
+        command: init
+        workingDirectory: $(Agent.BuildDirectory)/s/azure/terraform
```

Comment:
Its cleaner to upload an artifact after build stage and download it in the deploy stage. This assumes the build agent is same. but if in the future we need to change different build agent at different stage this could break.